packages/cactus-plugin-keychain-google-sm/src/main/typescript/plugin-keychain-google-sm-remote-adapter.ts

```diff
@@ -0,0 +1,151 @@
+ * plugin instance of the keychain plugin or an adapter backed by an API client
+ * object in which scenario the real keychain plugin object can be anywhere
+ * else on the network and also can be written in any programming language that
+ * the author so desires.
+ */
+export class PluginKeychainGoogleSmRemoteAdapter
+  implements ICactusPlugin, IPluginWebService {
+  public static readonly CLASS_NAME = "PluginKeychainGoogleSmRemoteAdapter";
+
+  private readonly instanceId: string;
+  private readonly keychainId: string;
+  private readonly log: Logger;
+  private readonly backend: DefaultApi;
+
+  public get className(): string {
+    return PluginKeychainGoogleSmRemoteAdapter.CLASS_NAME;
+  }
+
+  constructor(
+    public readonly opts: IPluginKeychainGoogleSmRemoteAdapterOptions,
+  ) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.keychainId, `${fnTag} arg options.keychainId`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    Checks.nonBlankString(opts.keychainId, `${fnTag} options.keychainId`);
+    Checks.truthy(opts.backend, `${fnTag} options.backend`);
+    Checks.truthy(
+      opts.backend instanceof DefaultApi,
+      `${fnTag} opts.backend instanceof DefaultApi`,
+    );
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = this.opts.instanceId;
+    this.backend = opts.backend;
+    this.keychainId = opts.keychainId;
+
+    this.log.info(`Created ${this.className}. KeychainID=${opts.keychainId}`);
+  }
+
+  /**
+   * Dummy implementation that wires no web services on the host API server
+   * because there is no need. All the functionality is implemented somewhere
+   * else on a host that's accessible through the network to this object
+   * (because this class is a remote adapter not an actual plugin impl.).
+   *
+   * @param _expressApp
+   */
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    return [];
+  }
+
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  public registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    return this.getOrCreateWebServices();
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.empty();
+  }
+
+  public async shutdown(): Promise<void> {
+    return;
+  }
+
+  public rotateEncryptionKeys(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getEncryptionAlgorithm(): string {
+    throw new Error("Method not implemented.");
+  }
+
+  public getKeychainId(): string {
+    return this.keychainId;
+  }
+
+  public async has(key: string): Promise<boolean> {
+    try {
+      await this.backend.getKeychainEntryV1({ key });
+      return true;
+    } catch (ex) {
+      // FIXME check for errors being thrown due to something other than
```

Comment:
@jagpreetsinghsasan Please narrow it down to only the exception that gets thrown when the key is not present

Reply:
I didnt write this method, because the getKeychainEntryV1( ) and similar methods aren't implemented and tested yet, in this PR. (Forgot to created the issue for this and the readme)

Created now
https://github.com/hyperledger/cactus/issues/1097 (get,set,delete keychainEntry endpoints) (point 3 of acceptance criteria specifically for this review comment)
https://github.com/hyperledger/cactus/issues/1096 (For Readme.md)